package/src/lib/helpers/guards.ts

```diff
@@ -6,7 +6,9 @@ import type {
   LowInclusiveBetweenOptions,
 } from "../NumberAssertion";
 
-export function isJSObject<T>(value: T): value is { [K in keyof T]: T[K] } {
+import { Struct } from "./types";
+
+export function isJSObject<T>(value: T): value is { [K in keyof T]: T[K] } & Struct {
   return value !== null
     && typeof value === "object"
     && typeof value !== "function";
```

Comment:
Just a question (outside of the scope of this pr): Is line 14 needed here? I think that checking for `typeof value === "object"` already implies it is not a function so maybe that line is not needed?

Reply:
Good catch @ChristianSama! Actually, the case we're missing here is to check the value is not an array because in JavaScript, arrays are also of type `object`. I'll fix that and add some tests to it 🙂